src/UI/Art/SpriteArtist/MiniMapArtist.java

```diff
@@ -89,29 +91,28 @@ private static BufferedImage generateFullMapImage(MapInfo mapInfo, Color[] teamC
       {
         XYCoord coord = new XYCoord(x, y);
         COSpriteSpec spec = null;
-        // Figure out team color, if any
+
+        boolean anyUnits = (mapInfo.mapUnits.size() > 0);
+        String unitName = null;
+        COSpriteSpec unitSpec = null;
+        boolean unitFlip = false, unitBuff = false;
+
         for( int co = 0; co < mapInfo.COProperties.length; ++co )
         {
+          // Figure out unit details, if any
+          if( anyUnits && mapInfo.mapUnits.get(co).containsKey(coord) )
+          {
+            unitName = mapInfo.mapUnits.get(co).get(coord);
+            unitSpec = getNthSpriteSpec(factions, teamColors, co);
+            unitFlip = (co % 2) == 1;
+          }
+
+          // Figure out team color, if any
           for( int i = 0; i < mapInfo.COProperties[co].length; ++i )
           {
             if( coord.equals(mapInfo.COProperties[co][i]) )
             {
-              Color coColor;
-
-              // Log a warning if SpriteLibrary doesn't have enough colors to support this map.
-              if( co >= teamColors.length )
-              {
-                System.out.println("WARNING! '" + mapInfo.mapName + "' has more start locations than there are team colors!");
-
-                // But soldier onwards anyway.
-                coColor = Color.BLACK;
```

Comment:
Are we just no longer worried about running out of colors, or is this case being handled in some other way?

Reply:
That logic is pushed into `getNthSpriteSpec()`